projects/aca-content/src/lib/components/search/search-results/search-results.component.ts

```diff
@@ -207,42 +204,55 @@ export class SearchResultsComponent extends PageComponent implements OnInit {
-        if (filtersFromEncodedQuery !== null) {
-          const filtersToLoad = this.queryBuilder.categories.length;
-          let loadedFilters = this.searchedWord === '' ? 0 : 1;
-          this.queryBuilder.filterLoaded
-            .asObservable()
-            .pipe(takeUntilDestroyed(this.destroyRef), takeUntil(this.loadedFilters$))
-            .subscribe(() => {
+      combineLatest([
+        this.route.queryParams,
+        this.router.events.pipe(
+          filter((e): e is NavigationStart => e instanceof NavigationStart),
+          startWith(null)
+        )
+      ])
+        .pipe(takeUntilDestroyed(this.destroyRef))
+        .subscribe(([params, navigationStartEvent]) => {
+          const shouldExecuteQuery = this.shouldExecuteQuery(navigationStartEvent, params);
+
+          this.savedSearchesService
+            .getSavedSearches()
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((savedSearches) => {
+              const savedSearchFound = savedSearches.find(
+                (savedSearch) => savedSearch.encodedUrl === encodeURIComponent(params[this.queryParamName])
+              );
+              this.initialSavedSearch = savedSearchFound !== undefined ? savedSearchFound : this.initialSavedSearch;
+            });
+          if (params[this.queryParamName]) {
+            this.isLoading = true;
+          }
+          this.loadedFilters$.next();
+          this.encodedQuery = params[this.queryParamName] || null;
+          this.searchedWord = extractSearchedWordFromEncodedQuery(this.encodedQuery);
+          this.updateUserQuery();
+          const filtersFromEncodedQuery = extractFiltersFromEncodedQuery(this.encodedQuery);
+          if (filtersFromEncodedQuery !== null) {
+            const filtersToLoad = this.queryBuilder.categories.length;
+            let loadedFilters = this.searchedWord === '' ? 0 : 1;
+            this.queryBuilder.filterLoaded.pipe(takeUntilDestroyed(this.destroyRef), takeUntil(this.loadedFilters$)).subscribe(() => {
               loadedFilters++;
               if (filtersToLoad === loadedFilters) {
                 this.loadedFilters$.next();
-                this.queryBuilder.execute(false);
+                if (shouldExecuteQuery) {
+                  this.queryBuilder.execute(false);
+                }
               }
             });
-          this.queryBuilder.populateFilters.next(filtersFromEncodedQuery);
-        } else {
-          this.queryBuilder.populateFilters.next({});
-          this.queryBuilder.execute(false);
-        }
-        this.queryBuilder.userQuery = extractUserQueryFromEncodedQuery(this.encodedQuery);
-      });
+            this.queryBuilder.populateFilters.next(filtersFromEncodedQuery);
+          } else {
+            this.queryBuilder.populateFilters.next({});
+            if (shouldExecuteQuery) {
+              this.queryBuilder.execute(false);
```

Comment:
similar warning like https://github.com/Alfresco/alfresco-content-app/pull/4666/files#r2196752122 for this line